compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -3057,24 +3093,21 @@ fn get_apple_sdk_root(sdk_name: &str) -> Result<String, errors::AppleSdkRootErro
     }
 }
 
-/// When using the linker flavors opting in to `lld`, add the necessary paths and arguments to
-/// invoke it:
+/// When using the linker flavors (or linker feature) opting in to `lld`, add the necessary paths
+/// and arguments to invoke it:
 /// - when the self-contained linker flag is active: the build of `lld` distributed with rustc,
 /// - or any `lld` available to `cc`.
+#[instrument(level = "debug", skip(cmd, sess))]
 fn add_lld_args(
```

Comment:
There's one more place where the distinction between lld and non-lld is important - `fn get_linker`, it will add `-flavor <FLAVOR>` to the command line if the linker is lld.
Right now it won't be added even if `-Clinker-features=+lld` is specified.

Ideally `fn linker_and_flavor` would return a flavor already adjusted for `-Clinker-features`, then everything would automatically work correctly after that.

Reply:
If that's the case then I'm not sure if we ever had `-flavor <FLAVOR>` for `-Zgcc-ld=lld` either, anything it did was way after `get_linker`.

Reply:
`-flavor <FLAVOR>` is not needed when lld is called through cc.
It's only needed when lld is called directly, and with an "improper" name not conveying the flavor, e.g. "rust-lld".